### ISSUE TITLE
Accessibility issues for <table> sorting and small viewport considerations

### ISSUE THREAD

**Issue by ericwbailey** [mdn/sprints#3840]
## URL(s)
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table#Table_sorting
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tableDisplaying_large_tables_in_small_spaces

## Request type

- [ ] New documentation
- [x] Correction or update

### Table sorting

The proposed technique for sorting table content will not work with many forms of assistive technology, notably screen readers and voice control. The sample JavaScript code does not communicate the presence of this intended interactivity, as table elements are used as clickable controls (see [WCAG Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) for more background).

Ideally, the sorting controls are implemented as [toggle buttons](https://www.w3.org/TR/wai-aria-practices-1.1/examples/button/button.html), with a label that indicates intended interactivity ("Sort alphabetically ascending", "Sort alphabetically descending", etc.).

### Displaying large tables in small spaces

The proposed technique for showing large amounts of table data in a small viewport involves updating the table's `display` property in CSS. This will destroy the native semantic mappings present in the table HTML element, meaning that assistive technology will loose the ability to recognize, describe, and navigate tabular data.

Some [more background on the issue](https://developer.paciellogroup.com/blog/2018/03/short-note-on-what-css-display-properties-do-to-table-semantics/), and [an accessible update to the proposed technique](https://adrianroselli.com/2017/11/a-responsive-accessible-table.html).

Thanks for considering, and thanks for all the work you do on such a lovely resource.

**Comment by chrisdavidmills** [comment]
Thanks for raising these concerns @ericwbailey .

@mattiapontonio and @brookjordan — you created these examples originally, and they are great, but I think Eric has some good points here — we really want our examples on MDN to be accessible wherever possible. Have you got some time to look at updating these?

I'd like to suggest that we hide the examples until the fixes are made. Are you OK with that? 

Best regards.

**Comment by mattiapontonio** [comment]
@chrisdavidmills I'm ok. I Will think at It.

**Comment by mattiapontonio** [comment]
Can We male a note about do not change display and wrap in a container for the responsive feature.
I have investigated this issue before. I appreciate but do not like the "vertical display" solution because of tables borns for comparing rows.
So i think wrapping Is a good solution.
For the buttons of course It Is right to use a button for funtionalities.

Responsive
https://www.w3schools.com/howto/howto_css_table_responsive.asp

I still not updated code.

**Comment by chrisdavidmills** [comment]
Thanks for looking into this @mattiapontonio . I've hidden the examples for now, but you'll still see them in the page source.

Sounds like the change for the sortable tables should be fairly easy. I'm not so sure about the large table in small space solution.

**Comment by mattiapontonio** [comment]
Is ok to put buttons in tables standard speaking? I've to check in validator.

**Comment by chrisdavidmills** [comment]
@mattiapontonio I don't know, I'm afraid.

**Comment by mattiapontonio** [comment]
No time to this effort now.
